common/src/main/java/com/revenuecat/purchases/common/BackendHelper.kt

```diff
@@ -0,0 +1,63 @@
+package com.revenuecat.purchases.common
+
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.common.networking.Endpoint
+import com.revenuecat.purchases.common.networking.HTTPResult
+import com.revenuecat.purchases.common.networking.RCHTTPStatusCodes
+import org.json.JSONObject
+
+class BackendHelper(
+    private val apiKey: String,
+    private val dispatcher: Dispatcher,
+    private val appConfig: AppConfig,
+    private val httpClient: HTTPClient
+) {
+    internal val authenticationHeaders = mapOf("Authorization" to "Bearer ${this.apiKey}")
+
+    fun performRequest(
+        endpoint: Endpoint,
+        body: Map<String, Any?>?,
+        onError: (PurchasesError) -> Unit,
+        onCompleted: (PurchasesError?, Int, JSONObject) -> Unit
+    ) {
+        enqueue(object : Dispatcher.AsyncCall() {
+            override fun call(): HTTPResult {
+                return httpClient.performRequest(
+                    appConfig.baseURL,
+                    endpoint,
+                    body,
+                    authenticationHeaders
+                )
+            }
+
+            override fun onError(error: PurchasesError) {
+                onError(error)
+            }
+
+            override fun onCompletion(result: HTTPResult) {
+                val error = if (!result.isSuccessful()) {
+                    result.toPurchasesError().also { errorLog(it) }
+                } else {
+                    null
+                }
+                onCompleted(error, result.responseCode, result.body)
```

Comment:
No need to change this now, since this is just moving this method's location, but we could reroute the errors to the `onError` callback instead of the `onCompleted` one, same way we do in other similar methods.

Reply:
Hmm interesting. I checked why we would be doing something like that and it looks like in `postSubscriberAttributes` we are using the `responseCode` in the onCompleted to determine the type of error. Which we could refactor... I don't want to break anything though so I think I am going to avoid doing that refactor now, since it would mean changing tests and fixing unit tests and I am behind with the offline entitlements feature. If that's fine of course

Reply:
Sure thing! No hurry on this